constrained_by_test.go

```diff
@@ -25,16 +27,152 @@ func (s *Zuite) TestWorksheet_constrainedBy() {
 	require.Equal(s.T(), `"Alex"`, ws.MustGet("name").String())
 }
 
-func (s *Zuite) TestWorksheet_constrainedByNonsensicalExpression() {
-	defs, err := NewDefinitions(strings.NewReader(`worksheet constrained_nonsensical_constrained_expression {
+func (s *Zuite) TestWorksheet_constrainedByNonBoolExpression() {
+	defs, err := NewDefinitions(strings.NewReader(`worksheet constrained_non_bool_constrained_expression {
 			69:some_field number[0] constrained_by { return some_field + 2 }
```

Comment:
Quick thing: it would be clearer to simply `return 5`, always try to have the minimal use case when writing tests.